masq/src/commands/financials_command.rs

```diff
@@ -58,26 +161,813 @@ impl Command for FinancialsCommand {
+            if amount_max.is_empty() {
+                Default::default()
+            } else {
+                apply_care(amount_max)
+            },
+        ]
+        .into_iter()
+        .map(assemble_optional_strings_into_single_number)
+        .collect::<VecDeque<String>>();
+        compose_ranges(vec_of_correct_values)
+    }
+
+    fn no_records_found<F>(stdout: &mut dyn Write, headings: &[&str], write_headings: F)
+    where
+        F: Fn(&mut dyn Write, &[(&str, &usize)]),
+    {
+        let headings_widths = Self::widths_of_str_values(headings);
+        write_headings(stdout, &Self::zip_them(headings, &headings_widths));
+        short_writeln!(stdout, "\nNo records found",)
+    }
+
+    fn figure_out_max_widths(values_of_accounts: &[Vec<String>]) -> Vec<usize> {
+        //two-dimensional set of strings; measuring their lengths and saving the largest value for each column
+        let column_count = values_of_accounts[0].len();
+        let init = vec![0_usize; column_count];
+        let widths_except_ordinal_num = values_of_accounts.iter().fold(init, |acc, record| {
+            Self::yield_bigger_values_from_vecs(
+                column_count,
+                acc,
+                Self::widths_of_str_values(record),
+            )
+        });
+        let mut result = vec![values_of_accounts.len().to_string().len()];
+        result.extend(widths_except_ordinal_num);
+        result
+    }
+
+    fn yield_bigger_values_from_vecs(
+        column_count: usize,
+        first: Vec<usize>,
+        second: Vec<usize>,
+    ) -> Vec<usize> {
+        (0..column_count).fold(vec![], |acc_inner, idx| {
+            plus(acc_inner, first[idx].max(second[idx]))
+        })
+    }
+
+    fn triple_or_single_blank_line(stdout: &mut dyn Write, leading_dump: bool) {
+        if leading_dump {
+            short_writeln!(stdout)
+        } else {
+            short_writeln!(stdout, "\n\n")
+        }
+    }
+
+    fn parse_integer<N: FromStr<Err = ParseIntError>>(str_num: &str) -> Result<N, String> {
+        str::parse::<N>(str_num).map_err(|e| match e.kind() {
+            IntErrorKind::NegOverflow | IntErrorKind::PosOverflow => panic!(
+                "Broken code: Clap validation should have caught this overflow of {} earlier",
+                str_num
+            ),
+            _ => format!(
+                "Non numeric value > {} <, all must be valid numbers",
+                str_num
+            ),
+        })
+    }
+
+    fn parse_time_params(time_range: &[&str]) -> Result<(u64, u64), String> {
+        Ok((
+            Self::parse_integer(time_range[0])?,
+            Self::parse_integer(time_range[1])?,
+        ))
+    }
+
+    fn single_capture(captures: &Captures, idx: usize) -> Option<String> {
+        captures.get(idx).map(|catch| catch.as_str().to_owned())
+    }
+
+    fn masq_values_range_regex() -> Regex {
+        Regex::new("(^(-?\\d+\\.?\\d*)\\s*-\\s*(-?\\d+\\.?\\d*))|(^-?\\d+\\.?\\d*)$")
```

Comment:
Probably either the ^s could be put outside all the groups, or the $ could be duplicated and put inside the groups.